src/elastio-snap.c

```diff
@@ -3381,6 +3386,13 @@ static int find_orig_mrf(struct block_device *bdev, make_request_fn **mrf){
 #else
 		// Linux versions older than 5.8
 		*mrf = q->make_request_fn;
+#ifdef HAVE_BLK_MQ_MAKE_REQUEST
+		if (!*mrf)
+		{
+			*mrf = elastio_snap_null_mrf;
+			LOG_DEBUG("original mrf is empty, set to elastio_snap_null_mrf with blk_mq_make_request");
+		}
+#endif        
```

Comment:
The comment above `// Linux versions older than 5.8` seems to be a quite misleading now, because  `HAVE_BLK_MQ_MAKE_REQUEST` is present on 5.8. And there are 2 very similar checks above and below. The one below is exactly the same, for the kernel 5.8. It can be used for the both 5.8 and  4.18-0.305 kernels.
Entire `find_orig_mrf` function became complicated and can be simplified.

Also same suggestion as in the comment above, to replace all usages of the `HAVE_BLK_ALLOC_QUEUE_MK_REQ_FN_NODE_ID` with the `HAVE_BLK_MQ_MAKE_REQUEST`here and in other applicable places.
 
Actually these 2 suggestions are not fixes for something, but will put defines in order and improve code readability.